MapboxVision/Services/DataProvider.swift

```diff
@@ -37,13 +40,21 @@ final class RecordedDataProvider: DataProvider {
         startTime = DispatchTime.now().uptimeMilliseconds
     }
     
+    private func getDelta() -> UInt {
+        let currentTime = DispatchTime.now().uptimeMilliseconds
+        let dt = DispatchTime.now().uptimeMilliseconds - startTime
+        startTime = currentTime
+        return dt
+    }
+    
     func update() {
         let settings = dependencies.recordingPath.settings
         let frameSize = Point2I(x: settings.width, y: settings.height)
-        let currentTimeMS = DispatchTime.now().uptimeMilliseconds - startTime + dependencies.startTime
-        telemetryPlayer.setCurrentTime(currentTimeMS)
+
+        time += getDelta()
```

Comment:
Seems like `DispatchTime.now().uptimeMilliseconds - startTime` is the delta since playback start so we don't have to calculate and store it additionally. Could you please test if that would work?